apps/examples/connect_test/connect_test_client.c

```diff
@@ -0,0 +1,676 @@
+	CT_LOG(TAG, "bssid %02x:%02x:%02x:%02x:%02x:%02x",
+		   msg.bssid[0], msg.bssid[1],
+		   msg.bssid[2], msg.bssid[3],
+		   msg.bssid[4], msg.bssid[5]);
+	WO_TEST_SIGNAL(WO_CONN_SUCCESS, g_wo_queue);
+}
+
+void wm_cb_softap_sta_leave(wifi_manager_cb_msg_s msg, void *arg)
+{
+	CT_LOG(TAG, "--> res(%d) reason %d", msg.res, msg.reason);
+	WO_TEST_SIGNAL(WO_CONN_SUCCESS, g_wo_queue);
+}
+
+void wm_cb_scan_done(wifi_manager_cb_msg_s msg, void *arg)
+{
+	CT_LOG(TAG, "--> res(%d)", msg.res);
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted soon eventually as you exit this function.
+	 */
+	if (msg.res != WIFI_MANAGER_SUCCESS || msg.scanlist == NULL) {
+		WO_TEST_SIGNAL(WO_CONN_FAIL, g_wo_queue);
+		return;
+	}
+	wt_print_scanlist(msg.scanlist);
+	WO_TEST_SIGNAL(WO_CONN_SUCCESS, g_wo_queue);
+}
+
+static void wm_get_created_softapinfo(wifi_manager_ap_config_s *ap_config)
+{
+	strncpy(ap_config->ssid, g_WM_SOFTAP_SSID, strlen(g_WM_SOFTAP_SSID) + 1);
+	ap_config->ssid_length = strlen(g_WM_SOFTAP_SSID);
+	ap_config->ap_auth_type = g_WM_AP_AUTH;
+	if (g_WM_AP_AUTH != WIFI_MANAGER_AUTH_OPEN) {
+		strncpy(ap_config->passphrase, g_WM_SOFTAP_PASSWORD, strlen(g_WM_SOFTAP_PASSWORD) + 1);
+		ap_config->passphrase_length = strlen(g_WM_SOFTAP_PASSWORD);
+		ap_config->ap_crypto_type = g_WM_AP_CRYPTO;
+	}
+}
+
+//UDP Data transfer
+static int _udp_send_data(int fd, int size, struct sockaddr_in* saddr)
+{
+	char buf[WT_BUF_SIZE];
+	int remain = size;
+	int slen = sizeof(struct sockaddr_in);
+	int cnt = 1;
+	int total_sent = 0; 
+	CT_LOG(TAG, "_udp_send_data start");
+	while (remain > 0) {
+		int send_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+		int nbytes = sendto(fd, buf, send_size, 0, (struct sockaddr*)saddr, slen);
+		if (nbytes == 0) {
+			CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+			CT_LOGE(TAG, "connection closed");
+			return -1;
+		} else if (nbytes < 0) {
+			if (errno == EWOULDBLOCK) {						 
+				CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+				CT_LOGE(TAG, "timeout error %d", errno);
+				return -1;
+			} else {
+				CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+				CT_LOGE(TAG, "connection error %d", errno);
+				return -1;
+			}
+		}
+		total_sent += nbytes;
+		remain -= nbytes;
+		cnt++;
+	}
+	CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+	return 0;
+}
+
+static int _udp_recv_data(int fd, int size, struct sockaddr_in* saddr)
+{
+	char buf[WT_BUF_SIZE];
+	unsigned int slen = sizeof(struct sockaddr_in);
+	int remain = size;
+	int cnt = 1;
+	int total_received = 0;
+	CT_LOG(TAG, "_udp_recv_data start");
+	while (remain > 0) {
+		int read_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+		int nbytes = recvfrom(fd, buf, read_size, 0, (struct sockaddr*)saddr, &slen);
+		if (nbytes == 0) {
+			CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+			CT_LOGE(TAG, "connection closed");
+			return -1;
+		} else if (nbytes < 0) {
+			if (errno == EWOULDBLOCK) {
```

Comment:
This'll be occurred if queue in network stack is empty. So it can try again after queue has space. 
If I'm misunderstood the situation please comment it in code

Reply:
In updated connect test app this comment is taken care.